lib/js/src/manager/screen/utils/SoftButtonObject.js

```diff
@@ -239,6 +243,144 @@ class SoftButtonObject {
         this._onEventListener = onEventListener;
         return this;
     }
+
+    /**
+     * Creates a deep copy of the object
+     * @returns {SoftButtonObject} deep copy of the object
+     */
+    clone () {
+        const statesClone = this.getStates().map((state) => {
+            const artwork = state.getArtwork();
+            let artworkClone = null;
+            if (artwork !== null && artwork !== undefined) {
+                const jsonClone = JSON.parse(JSON.stringify(artwork));
+                artworkClone = new SdlArtwork()
+                    .setName(jsonClone._fileName)
+                    .setFilePath(jsonClone._filePath)
+                    .setFileData(jsonClone._fileData)
+                    .setType(jsonClone._fileType)
+                    .setPersistent(jsonClone._persistentFile)
+                    .setStaticIcon(jsonClone._isStaticIcon)
+                    .setTemplateImage(jsonClone._isTemplate)
+                    .setOverwrite(jsonClone._overwrite);
+            }
+            const buttonState = new SoftButtonState(
+                JSON.parse(JSON.stringify(state.getName())),
+                JSON.parse(JSON.stringify(state.getSoftButton().getText())),
+                artworkClone
+            );
+
+            buttonState._softButton = new SoftButton(state.getSoftButton().getParameters());
```

Comment:
This line does not deep copy the SoftButton parameters object and so the clone's soft button parameters will have the same reference as the original

Reply:
Stringifying the parameters wouldn't work here because the Image param is a Struct which would need to be cloned itself making the logic a little messier. I reverted this back to the way it was before.